File: ruby/trailing_zeroes/counting_trailing_zeroes_spec.rb
require 'minitest/autorun'

module TrailingZeroes
  extend self

  def call(number)
    pow = 1
    amount = 0
    while ( (number / (5**pow) > 0 ) )
      amount += (number / (5**pow))
      pow += 1
    end
    amount
  end
end

describe TrailingZeroes do
  it 'calculates the trailing zeroes for the 12' do
    TrailingZeroes.(12).must_equal 2
  end

  it 'calculates the trailing zeroes for the 1000' do
    TrailingZeroes.(1000).must_equal 249
  end
end

File: javascript/counting_trailing_zeroes/test/counting_trailing_zeroes_spec.js
var expect = require("chai").expect;

var countTrailingZeroes = function(number) {
  var pow = 1;
  var amount = 0;
  while ((number / Math.pow(5, pow)) > 1) { 
    amount += (number / Math.pow(5, pow));
    pow += 1;
  }
  return Math.floor(amount);
}

describe("Counting trailing zeroes", function() { 
  it("calculates the trailing zeroes for 12", function() { 
    expect(countTrailingZeroes(12)).to.equal(2);
  });

  it("calculates the trailing zeroes for 1000", function() { 
    expect(countTrailingZeroes(1000)).to.equal(249);
  });
});
